src/client/components/ToggleSwitchInput.tsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import type { SerializedStyles } from '@emotion/react';
+import type { EmotionJSX } from '@emotion/react/types/jsx-namespace';
+import type { Props } from '@guardian/source-react-components';
+import { css } from '@emotion/react';
+import {
+  neutral,
+  success,
+  textSans,
+  focusHalo,
+  transitions,
+  descriptionId,
+  generateSourceId,
+} from '@guardian/source-foundations';
+
+const inputStyles = css`
+  opacity: 0;
+  position: absolute;
+  top: 0;
+  left: 0;
+`;
+
+const labelStyles = css`
+  user-select: none;
+  position: relative;
+  ${textSans.small()};
+  display: flex;
+  color: ${neutral[7]};
+  align-items: center;
+  cursor: pointer;
+`;
+
+const switchStyles = (labelPosition: LabelPosition) => css`
+  flex: none;
+  border: none;
+  margin: ${labelPosition === 'left' ? '0px 0px 0px 8px' : '0px 8px 0px 0px'};
+  padding: 0;
+  display: inline-block;
+  text-align: center;
+  position: relative;
+  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
+  cursor: pointer;
+
+  &:after {
+    content: '';
+    position: absolute;
+    border-radius: 50%;
+    background: #fff;
+    will-change: left;
+    transition: left 0.15s ease-in-out;
+  }
+`;
+
+const siblingStyles = css`
+  input + span {
+    background-color: rgba(153, 153, 153, 0.5);
+  }
+
+  /* TODO focus-visible not fully supported */
+  input:focus-visible + span {
```

Comment:
Struggling to fine a better solution to this (showing outline on tab focus but not mouse hover/click)